Library/Homebrew/development_tools.rb

```diff
@@ -63,23 +67,27 @@ def gcc_42_build_version
     def clang_version
       @clang_version ||=
         if (path = locate("clang"))
-          `#{path} --version`[/(?:clang|LLVM) version (\d\.\d)/, 1]
+          Version.new `#{path} --version`[/(?:clang|LLVM) version (\d\.\d)/, 1]
+        else
+          Version::NULL
         end
     end
 
     def clang_build_version
       @clang_build_version ||=
         if (path = locate("clang"))
-          `#{path} --version`[/clang-(\d{2,})/, 1].to_i
+          Version.new `#{path} --version`[/clang-(\d{2,})/, 1]
+        else
+          Version::NULL
         end
     end
 
     def non_apple_gcc_version(cc)
       (@non_apple_gcc_version ||= {}).fetch(cc) do
         path = HOMEBREW_PREFIX.join("opt", "gcc", "bin", cc)
         path = locate(cc) unless path.exist?
-        version = `#{path} --version`[/gcc(?:-\d(?:\.\d)? \(.+\))? (\d\.\d\.\d)/, 1] if path
-        @non_apple_gcc_version[cc] = version
+        version = Version.new(`#{path} --version`[/gcc(?:-\d(?:\.\d)? \(.+\))? (\d\.\d\.\d)/, 1]) if path
```

Comment:
I know it's not your code but: sticking the `if` in the block form would make it more obvious given the line length.